app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -262,6 +269,29 @@ def remove_disk_config_spec(vim_obj, vmcs, hardware, options)
     end
   end
 
+  def resize_disk_config_spec(vim_obj, vmcs, hardware, options)
+    raise "resize_disk_config_spec: disk filename is required." unless options[:disk_name]
+
+    device = vim_obj.getDeviceByBacking(options[:disk_name], hardware)
+    raise "resize_disk_config_spec: no virtual device associated with: #{options[:disk_name]}" unless device
+    raise "resize_disk_config_spec: decrease size is not supported for: #{options[:disk_name]}" unless device.capacityInKB.to_i <= options[:disk_size_in_mb].to_i * 1024
+
+    add_device_config_spec(vmcs, VirtualDeviceConfigSpecOperation::Edit) do |vdcs|
+      vdcs.device = VimHash.new("VirtualDisk") do |dev|
+        dev.key           = device.key
+        dev.capacityInKB  = options[:disk_size_in_mb].to_i * 1024
+        dev.controllerKey = device.controllerKey
+        dev.unitNumber    = device.unitNumber
+
+        dev.backing = VimHash.new("VirtualDiskFlatVer2BackingInfo") do |bck|
+          bck.diskMode        = device.backing.diskMode
+          bck.thinProvisioned = device.backing.thinProvisioned
+          bck.fileName        = device.backing.fileName
```

Comment:
These are tricky because they aren't defined for all backing types, not even all types that extend `VirtualDeviceFileBackingInfo`.  E.g. `VirtualDiskRawDiskMappingVer1BackingInfo` doesn't have `thinProvisioned` or `fileName`.

If we _have_ to send a backing I would prefer sending the entire backing from the existing disk so we don't have to worry about these issues.

Reply:
Yes without the backing the call fails. SO I think something like
```
dev.backing = device.backing
```
I will try this tomorrow.